src/main/scala/com/cognite/sdk/scala/v1/propertyMap.scala

```diff
@@ -57,8 +103,61 @@ final case class DataModelInstanceQueryResponse(
     modelProperties: Option[Map[String, DataModelPropertyDefinition]] = None,
     nextCursor: Option[String] = None
 )
+object DataModelInstanceQueryResponse {
+  def createDecoderForQueryResponse(): Decoder[DataModelInstanceQueryResponse] = {
+    import DataModels.dataModelPropertyDefinitionDecoder
+
+    (c: HCursor) =>
+      for {
+        modelProperties <- c
+          .downField("modelProperties")
+          .as[Map[String, DataModelPropertyDefinition]]
+
+        seqDecoder: Decoder[Seq[PropertyMap]] =
+          Decoder.decodeIterable[PropertyMap, Seq](
+            PropertyMap.createDynamicPropertyDecoder(modelProperties),
+            implicitly
+          )
+
+        items <- seqDecoder.tryDecode(c.downField("items"))
+        nextCursor <- c.downField("nextCursor").as[Option[String]]
+      } yield DataModelInstanceQueryResponse(items, Some(modelProperties), nextCursor)
+  }
+}
 
 final case class DataModelInstanceByExternalId(
     items: Seq[CogniteExternalId],
     model: DataModelIdentifier
 )
+
+final case class Edge(
+    override val externalId: String,
+    `type`: String,
+    startNode: String,
+    endNode: String,
+    properties: Option[Map[String, DataModelProperty[_]]] = None
+) extends PropertyMap(
+      {
+        val propsToAdd: Seq[Option[(String, DataModelProperty[_])]] =
+          Seq[(String, Option[DataModelProperty[_]])](
+            "externalId" -> Some(PropertyType.Text.Property(externalId)),
+            "type" -> Some(PropertyType.Text.Property(`type`)),
+            "startNode" -> Some(PropertyType.Text.Property(startNode)),
+            "endNode" -> Some(PropertyType.Text.Property(endNode))
+          ).map { case (k, v) =>
+            v.map(k -> _)
+          }
```

Comment:
Why so complicated, can we just do like this and without the `.flatten.toMap` on `propsToAdd` later instead ? 😅 
```suggestion
        val propsToAdd: Map[String, DataModelProperty[_]] = Map(
          "externalId" -> PropertyType.Text.Property(externalId),
          "type" -> PropertyType.Text.Property(`type`),
          "startNode" -> PropertyType.Text.Property(startNode),
          "endNode" -> PropertyType.Text.Property(endNode)
        )
```